article/app/controllers/ArticleController.scala

```diff
@@ -234,15 +260,29 @@ class ArticleController extends Controller with RendersItemResponse with Logging
 }
 
 object ParseBlockId extends RegexParsers {
-  def apply(input: String): Option[String] = {
-    def withParser: Parser[Unit] = "with:" ^^ { _ => () }
-    def block: Parser[Unit] = "block-" ^^ { _ => () }
-    def id: Parser[String] = "[a-zA-Z0-9]+".r
-    def expr: Parser[String] = withParser ~> block ~> id
+
```

Comment:
yeah in the grand scheme of this PR it won't be much (it happens once per request) but Phil W pointed me at parboiled for something else so I'll be using it in future.  I'll add it to my list of tweaks to do once I've got the big perf stuff out of the way as it's a good idea.